frontend/frontend-filters-links.php

```diff
@@ -407,6 +407,13 @@ public function check_canonical_url( $requested_url = '', $do_redirect = true )
 			if ( ! empty( $obj ) && $this->model->is_translated_taxonomy( $obj->taxonomy ) ) {
 				$language = $this->model->term->get_language( (int) $obj->term_id );
 			}
+
+			if ( is_feed() && empty( $obj ) ) { // Allows to replace the language correctly in a category feed query.
+				if ( $this->model->is_translated_taxonomy( $this->get_queried_taxonomy( $this->wp_query()->tax_query ) ) ) {
+					$term_id = $this->get_queried_term_id( $this->wp_query()->tax_query );
+					$language = $this->model->term->get_language( $term_id );
+				}
```

Comment:
I suggest to avoid the repetition of the same code below.

Reply:
Indeed, I will refactor.